pkgs/tools/misc/goose/default.nix

```
@@ -0,0 +1,43 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "goose";
+  version = "3.11.2";
+
+  src = fetchFromGitHub {
+    owner = "pressly";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-D/mWzDsGNftl0ldNpWt/98b2o5fWk4Jxb5zEIAqTPs0=";
+  };
+
+  vendorHash = "sha256-n5aUV+FXjfAmzToRH/v4Rq9/sUkmVaSvo32058q/3nE=";
+
+  # end-to-end tests require a docker daemon
+  postPatch = ''
+    rm -r tests/e2e
+    rm -r tests/gomigrations
+    rm -r tests/vertica
+  '';
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X=main.gooseVersion=${version}"
+  ];
+
+  checkFlags = [
+    # these also require a docker daemon
+    "-skip=TestClickUpDown|TestClickHouseFirstThree"
+  ];
+
```

I couldn't figure out why or what tests are failing on darwin

```suggestion

  doCheck = !stdenv.isDarwin;

```